0000-sign-miner.md

```diff
@@ -0,0 +1,159 @@
+be tagged and signed by the third party before they can be deployed to the fleet
+of consumer Hotspots. This increased overhead could slow the pace of feature
+development and delay the release of critical bug fixes.
+
+The Buildroot embedded Linux build system that nextgate relies on to build a
+Hotspot firmware image already uses Git to clone the contents of the
+helium/miner repo at a given Git commit hash. Helium's Embedded Engineering
+team will need to replace the repo site in the miner makefile with the URL of
+the third party's miner repo. Every time there is a new miner release the
+Helium Engineering team will update the desired miner version with the latest
+miner release tag. A Git tag verification step will need to be added to the
+miner package makefile to ensure that the release tag verifies against the third
+party's GPG public key.
+
+# Rationale and Alternatives
+[alternatives]: #rationale-and-alternatives
+
+There are three ways to sign a miner release:
+
+1. Sign a Git tag.
+2. Sign a source tarball of a Git tag.
+3. Sign a gzipped bundle of binaries built from a Git tag.
+
+Signing a Git tag is the easiest because Git is the only tool required by both
+the signer and verifier. Signing a source tarball is slightly harder because
+another tool like OpenBSD's signify needs to be provisioned for both signing and
+verifying. Source tarballs also necessitate checksum verification in case a
+download is truncated, corrupt or tampered with. Signing a gzipped bundle of
+binaries is the most difficult option because we have to wait for miner and all
+its dependencies to build for the target platform.
+
+- What is the impact of not doing this?
+
+If we do not separate miner from nextgate for signing we eventually risk greater
+scrutiny from Hotspot operators, regulators and large cryptocurrency exchanges.
+
+# Unresolved Questions
+[unresolved]: #unresolved-questions
+
+- What parts of the design do you expect to resolve through the HIP process
+before this gets merged?
+
+Which signing method integrates best with our existing build system and OTA
+update mechanism?
+
+- What related issues do you consider out of scope for this HIP that could be
+addressed in the future independently of the solution that comes out of this HIP?
+
+How will Hotspot owners deliver their own OTA updates securely?
+
+# Deployment Impact
+[deployment-impact]: #deployment-impact
+
+- How will current users be impacted?
+
+Consumer Hotspot owners will not notice a difference. DIY Hotspot makers and HNT
+cloud miners can choose to verify miner Git release tags as part of their
+software CI/CD pipeline if they feel so inclined.
+
+# Success Metrics
+[success-metrics]: #success-metrics
+
+Success is measured by:
+
+- time it takes to Helium Blockchain and Embedded teams to implement a signing
+and verification solution
+
+- effort required by Helium Blockchain team to issue a miner release
```

Comment:
From @amirhaleem 

> Is there a way that this could be architected such that the miner is an entirely separate OTA function than nextgate? So on a Helium Hotspot nextgate would be a mandatory update, but a user could opt to change where miner comes from and deal with the appropriate keys/signature verification for that repo (similar to adding a custom apt repo)? As a Hotspot owner I'd be able to configure (via BLE) which OTA server/keys I want to use for the miner part of the firmware.
> 
> My understanding of what you've written above is that a user still has no way of, for example, specifying their own miner release with modified code if they choose to.

Reply:
That's correct. How to a point a miner at homemade miner release is addressed by PR #11.